lib/rubocop/cop/rspec/excessive_docstring_spacing.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for excessive whitespace in example descriptions.
+      #
+      # @example
+      #   # bad
+      #   it '  has  excessive   spacing  ' do
+      #   end
+      #
+      #   # good
+      #   it 'has excessive spacing' do
+      #   end
+      #
+      # @example
+      #   # bad
+      #   context '  when a condition   is met  ' do
+      #   end
+      #
+      #   # good
+      #   context 'when a condition is met' do
+      #   end
+      class ExcessiveDocstringSpacing < Base
+        extend AutoCorrector
+
+        MSG = 'Excessive whitespace.'
+
+        # @!method example_description(node)
+        def_node_matcher :example_description, <<-PATTERN
+          (block (send _ {:describe :context :it} ${
+            (str $_)
+            $(dstr ({str dstr `sym} ...) ...)
+          } ...) ...)
+        PATTERN
+
+        def on_block(node)
+          example_description(node) do |description_node, message|
+            current_text = text(message)
+            correct_text = strip_excessive_whitespace(current_text)
```

Comment:
I'm not sure how much performance matters, but it seems we’ll be allocating a lot of extra string objects here. I believe it would be much faster with e.g.
```rb
if text.start_with?(' ') || text.include?('  ') || text.end_with?(' ')
  correct_text = strip_excessive_whitespace(current_text)
```

Reply:
Good point - I've done a slightly more complex refactor based on this, as I realised I can just do this check rather than comparing the stripped vs unstripped versions.